pkgs/by-name/cl/clash-verge-rev/package.nix

```diff
@@ -1,66 +1,179 @@
-    wrapGAppsHook3
-    autoPatchelfHook
-  ];
+  src-service = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-service";
+    rev = "e74e419f004275cbf35a427337d3f8c771408f07"; # no meaningful tags in this repo. The only way is updating manully every time.
+    hash = "sha256-HyRTOqPj4SnV9gktqRegxOYz9c8mQHOX+IrdZlHhYpo=";
+  };
 
-  buildInputs = [
-    openssl
-    webkitgtk
-  ];
+  service = rustPlatform.buildRustPackage {
+    pname = "${pname}-service";
+    inherit version;
 
-  runtimeDependencies = [
-    (lib.getLib udev)
-    libayatana-appindicator
-  ];
+    src = src-service;
+    sourceRoot = "${src-service.name}";
 
-  installPhase = ''
-    runHook preInstall
+    nativeBuildInputs = [
+      pkg-config
+      rustPlatform.cargoSetupHook
+    ];
 
-    mkdir -p $out/bin
-    mv usr/* $out
+    buildInputs = [
+      openssl
+      webkitgtk
+    ];
 
-    runHook postInstall
-  '';
+    env = {
+      OPENSSL_NO_VENDOR = 1;
+    };
 
-  postFixup = ''
-    rm -f $out/bin/verge-mihomo
-    ln -sf ${lib.getExe clash-meta} $out/bin/verge-mihomo
-  '';
+    cargoHash = "sha256-tjPGXsABNK3HR6QmVl97UqGkDBhqhL3sSc3vGScWmho=";
+  };
 
-  passthru.updateScript = nix-update-script { };
+  webui = stdenv.mkDerivation {
+    inherit version src;
+    pname = "${pname}-webui";
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-DYsx1X1yXYEPFuMlvZtbJdefcCR8/wSUidFwsMy8oLk=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm.configHook
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      node --max_old_space_size=1024000 ./node_modules/vite/bin/vite.js build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -r dist $out
+
+      runHook postInstall
+    '';
+  };
 
   meta = {
     description = "Clash GUI based on tauri";
     homepage = "https://github.com/clash-verge-rev/clash-verge-rev";
     license = lib.licenses.gpl3Plus;
     mainProgram = "clash-verge";
-    maintainers = with lib.maintainers; [ Guanran928 ];
+    maintainers = with lib.maintainers; [
+      Guanran928
+      bot-wxt1221
+    ];
     platforms = [ "x86_64-linux" ];
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
   };
-})
+in
+rustPlatform.buildRustPackage {
+  inherit
+    pname
+    version
+    src
+    meta
+    ;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo-tauri.lock;
+    outputHashes = {
+      "sysproxy-0.3.0" = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
+    };
+  };
+
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+  };
+
+  postPatch = ''
+    substituteInPlace ./tauri.conf.json \
+      --replace-fail '"distDir": "../dist",' '"distDir": "${webui}",' \
+      --replace-fail '"beforeBuildCommand": "pnpm run web:build"' '"beforeBuildCommand": ""'
+
+    mkdir -p ./sidecar
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-${rust.envVars.rustHostPlatform}
```

Comment:
I thought we should use something like

```
    substituteInPlace src/core/core.rs \
      --replace-fail 'new_sidecar(clash_core)?' 'new(clash_core)'
```

to make clash verge find core from path so that we don't need to hardcode the core. If the user want to use a different core they don't need to rebuild the whole clash-verge.